### ISSUE TITLE
GOADB spec

### ISSUE THREAD

**Issue by typemytype** [adobe-type-tools/afdko#682]
there is a spec for GOADB files in the [old user guide](https://github.com/adobe-type-tools/afdko/blob/develop/docs/MakeOTFUserGuide.pdf) but it seems there are some changed in the GOADB spec.

Could it be described somewhere in this repo?

seeing GOADB of [Source Serif](https://github.com/adobe-fonts/source-serif-pro/blob/master/GlyphOrderAndAliasDB#L2) double unicodes are possible now as a comma separated list (this is a guess and guessing specs is not so nice)

thanks

**Comment by readroberts** [comment]
You are correct the makeotf now supports specifying multiple Unicode values for a glyph, and and I see that unfortunately you are also correct that this is not documented anywhere. We will fix this.

FYI, the syntax is to use the same format of glyph name that you would use to specify a single Unicode value, and to concatenate the UV names as a comma-separated list, no whitespace allowed, and put this in the third field of a GOADB file record.

**Comment by typemytype** [comment]
thanks Read! please document ;)

**Comment by typemytype** [comment]
Read could you also specify if higher unicodes still needs a `u<hex>` notation and `uni<hex>` for hex values smaller or equal then `FFFF` also in the comma separated list?

fe `space	space	uni0020,uni00A0,u1F30C`

thanks

**Comment by frankrolf** [comment]
I have been talking a little to @typemytype about this (IRT #766), and I am unclear about the following scenarios:

- what happens if a friendly glyph name contains a hyphen but no Unicode (how to avoid clashes with existing glyph names?)
- what happens if the friendly glyph name with hyphen contains a PUA Unicode? Override in 3rd column? Same clashing problem as above I suppose?

**Comment by miguelsousa** [comment]
@frankrolf the best way to answer those questions is to build test fonts. If you're going to do it keep in mind the following:

* the results of running makeotf in release mode are different from running in development mode
* clashing of final glyph names is NOT a new problem. I know for a fact that when there are repeated final glyphs names one of the entries will prevail over the other(s); I just don't remember if it's the first or the last (BTW, the order is according to the GOADB, of course)
* #766 is about using invalid characters in final glyph names. I think it's best to treat that case separately from the questions you're trying to answer. By that I mean, first answer your questions without using invalid final glyph names. Once you have your answers, introduce invalid final glyph names in the GOADB and take conclusions based on that second round of results.
